jni/src/knn_extension/faiss/utils/BitSet.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include <algorithm>
+#include "BitSet.h"
+
+FixedBitSet::FixedBitSet(const int* int_array, const int length){
+    assert(int_array && "int_array should not be null");
+    const int* maxValue = std::max_element(int_array, int_array + length);
+    this->numBits = (*maxValue >> 6) + 1; // div by 64
```

Comment:
I dont think numBits is the right name -> numBits is really numBits/64. Its the number of ints in the bitmap

Reply:
You are right. Here the "word" is a bit block. It should be numWords. Let me update it. Thanks for correcting it.